napari/_vispy/vispy_image_layer.py

```diff
@@ -39,11 +40,13 @@ def _on_display_change(self, data=None):
         self.node.parent = None
 
         if self.layer.dims.ndisplay == 2:
-            self.node = ImageNode(data, method='auto')
+            self.node = self._image_node
+            self.node.set_data(data)
         else:
             if data is None:
                 data = np.zeros((1, 1, 1))
-            self.node = VolumeNode(data, clim=self.layer.contrast_limits)
+            self.node = self._volume_node
+            self.node.set_data(data, clim=self.layer.contrast_limits)
```

Comment:
@sofroniewn the only thing I would change here is to *first* set the data on the appropriate node, and *then* swap out the node. ie:

```python
self._image_node.set_data(data)
self.node = self._image_node
...
self._volume_node.set_data(data)
self.node = self._volume_node
```

To me that seems cleaner in that you never enter an inconsistent state, in some sense...

Other than that, since this seems to fix the issue for everyone, I suggest merging now and continue investigating in future PRs!

Reply:
done